Sources/NIO/Thread.swift

```diff
@@ -128,27 +128,33 @@ final class Thread {
 ///
 /// `ThreadSpecificVariable` is thread-safe so it can be used with multiple threads at the same time but the value
 /// returned by `currentValue` is defined per thread.
-///
-/// - note: `ThreadSpecificVariable` has reference semantics.
-public struct ThreadSpecificVariable<T: AnyObject> {
+public class ThreadSpecificVariable<T: AnyObject> {
```

Comment:
@weissi I think we also want to make this final

Reply:
done